src/server/system_services/system_server.js

```diff
@@ -1309,6 +1309,246 @@ async function _get_endpoint_groups() {
+            const reencrypted = mkm._reencrypt_master_key(bucket.master_key_id._id, new_master_key._id);
+            await upsert_master_key({ _id: bucket.master_key_id._id,
+                    update: { cipher_key: reencrypted, master_key_id: new_master_key._id}});
+        }));
+        await P.all(_.map(system_store.data.accounts, async function(account) {
+            if (account.email.unwrap() === "support@noobaa.com") return;
+            const reencrypted = mkm._reencrypt_master_key(account.master_key_id._id, new_master_key._id);
+            await upsert_master_key({ _id: account.master_key_id._id,
+                    update: { cipher_key: reencrypted, master_key_id: new_master_key._id}});
+        }));
+    }
+
+    if (entity_type === 'ACCOUNT') {
+        const new_s3_secret_keys_list = _.map(entity_info.access_keys, function(creds) {
+            creds.secret_key = mkm.encrypt_sensitive_string_with_master_key_id(creds.secret_key, new_master_key._id);
+            return creds;
+        });
+        const new_sync_creds_list = _.map(entity_info.sync_credentials_cache, function(creds) {
+            creds.secret_key = mkm.encrypt_sensitive_string_with_master_key_id(creds.secret_key, new_master_key._id);
+            return creds;
+        });
+        const pools_ns_resource_updates = get_pools_and_ns_resources_changes(new_sync_creds_list, entity_info._id);
+        await update_account_secrets(entity_info._id, new_s3_secret_keys_list, new_sync_creds_list, pools_ns_resource_updates);
+        // TODO: delete the old { encrypted secret key : decrypted secret key } pair from access keys LRU cache in master key manager
+    }
+}
+
+/**
+ *
+ * DISABLE MASTER KEY
+ *
+ */
+async function disable_master_key(req) {
+    const { entity, entity_type } = req.rpc_params; // System Name, Bucket Name, Account Email
+    await _disable_master_key(entity, entity_type);
+}
+
+// when disabling system, we need to disable the lower sub trees.
+// when disabling account - need to decrypt all the access keys.
+// when disabling bucket, the map builder bg worker will decrypt 
+// the cipher chunks and remove the master key id of the chunk.
+async function _disable_master_key(entity, entity_type) {
+    const entity_info = get_entity_info(entity, entity_type);
+    if (!entity_info) throw new Error(`_disable_master_key: entity doesn't exist ${entity}`);
+
+    if (!entity_info.master_key_id || entity_info.master_key_id.disabled === true) {
+        throw new Error(`_disable_master_key: can not disable master key, current master key is: ${util.inspect(entity_info)}`);
+    }
+
+    await upsert_master_key({ _id: entity_info.master_key_id._id, update: {disabled: true}});
+    system_store.master_key_manager.set_m_key_disabled_val(entity_info.master_key_id._id, true);
+
+    await system_store.load();
+    if (entity_type === 'ACCOUNT') {
+        const decrypted_s3_creds = entity_info.access_keys;
```

Comment:
This seems a bit weird.
We are updating the master_key ad then reloading the whole accounts.
After that, you take the encrypted values and do not perform decryption on them and just push them decrypted inside the DB.
Please check that and apply a fix to tests if needed.